source/postprocess/basic_statistics.cc

```diff
@@ -44,6 +44,9 @@ namespace aspect
               const MaterialModel::Simple<dim> &material_model
                 = dynamic_cast<const MaterialModel::Simple<dim> &>(this->get_material_model());
 
+              MaterialModel::MaterialModelInputs<dim> in (1, this->n_compositional_fields());
+              MaterialModel::MaterialModelOutputs out (1, this->n_compositional_fields());
```

Comment:
basic_statistics.cc constructs its own material model in order to query it for values later on. For some reason it casts the actual material model to a Simple material model, and then calls functions from it. I'm removing some of those functions from the MaterialModel::Simple class, so I need to somehow still be able to query it for values. I suspect that the dynamic cast is not important anymore, but I wanted to make minimal changes.